src/extensions/extension-discovery.ts

```diff
@@ -234,24 +234,28 @@ export class ExtensionDiscovery extends Singleton {
       return;
     }
 
-    const extension = Array.from(this.extensions.values()).find((extension) => extension.absolutePath === filePath);
+    for (const extension of this.extensions.values()) {
```

Comment:
This does not feel like an improvement.. why do we want `for` loop here?

Reply:
Mainly because it is wasteful the create an array just to pick an item out of it. 

If you prefer I could have used a theoretical `iter.find()` function instead.